pkgs/os-specific/linux/realmd/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib, stdenv
+, autoreconfHook
+, docbook_xml_dtd_412
+, docbook_xml_dtd_42
+, docbook_xml_dtd_43
+, docbook_xsl
+, fetchFromGitLab
+, glib
+, intltool
+, libkrb5
+, libtool
+, libxslt
+, openldap
+, pkg-config
+, polkit
+, systemd
+, xmlto
+}:
+# doc-related (--disable-doc): docbook_xml_dtd_412 docbook_xml_dtd_42 docbook_xml_dtd_43 docbook_xsl libxslt xmlto
+
+stdenv.mkDerivation rec {
+  pname = "realmd";
+  version = "0.17.0";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "1c6q2a86kk2f1akzc36nh52hfwsmmc0mbp6ayyjxj4zsyk9zx5bf";
+  };
+
+  configureFlags = [
+    "--with-distro=NixOS"
+    "--with-systemd-unit-dir=${placeholder "out"}/lib/systemd/system"
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    docbook_xml_dtd_412
+    docbook_xml_dtd_42
+    docbook_xml_dtd_43
+    docbook_xsl
+    glib
+    intltool
+    libkrb5
+    libtool
+    libxslt
+    openldap
+    pkg-config
+    polkit
+    systemd
+    xmlto
+  ];
+
+  preConfigure = ''
+    # service/Makefile.am expects to find the file "service/realmd-NixOS.conf", which:
+    # - isn't provided by upstream
+    # - doesn't make sense in this context and should rather be provided by a module configuring the realmd service
+    sed -i /DISTRO/d service/Makefile.am
+    # don't install "/var/cache" and "/var/lib" - this is provided by the service unit at runtime
+    sed -i '/INSTALL_PROGRAM.\+cachedir/d' service/Makefile.am
+    sed -i '/INSTALL_PROGRAM.\+localstatedir/d' service/Makefile.am
+  '';
+
+  meta = with lib; {
+    homepage = "https://gitlab.freedesktop.org/realmd/realmd";
+    description = "Service + helpers for configuring Kerberos, AD-membership and other online identities";
+    platforms = platforms.linux;
+    license = licenses.lgpl21;
```

Comment:
Do you know if the license is lgpl21Only or lgpl21Plus?

Reply:
Looks like `lgpl21Plus` to me, as:
- the [corresponding commit](https://gitlab.freedesktop.org/realmd/realmd/-/commit/e915a592d4ee3aaf98c62a4dd8c1d4b216a4291f) states:
  > Update license to LGPL v2.1
  > 
  > Since parts of the code are LGPL v2.1 or later, while most of the code
  > is LGPL v2.0 or later, update the overall license be LGPL v2.1.
  > 
  > https://bugs.freedesktop.org/show_bug.cgi?id=57943
- the [license itself](https://gitlab.freedesktop.org/realmd/realmd/-/blob/master/COPYING) states:
  > Each version is given a distinguishing version number.  If the Library
  > specifies a version number of this License which applies to it and
  > "any later version", you have the option of following the terms and
  > conditions either of that version or of any later version published by
  > the Free Software Foundation.  If the Library does not specify a
  > license version number, you may choose any version ever published by
  > the Free Software Foundation.

  and…

  > This library is free software; you can redistribute it and/or
  > modify it under the terms of the GNU Lesser General Public
  > License as published by the Free Software Foundation; either
  > version 2.1 of the License, or (at your option) any later version.